src/dbt_client/dbtIntegration.ts

```diff
@@ -39,27 +40,34 @@ export class CLIDBTCommandExecutionStrategy
   implements DBTCommandExecutionStrategy
 {
   constructor(
-    private commandProcessExecutionFactory: CommandProcessExecutionFactory,
-    private pythonEnvironment: PythonEnvironment,
-    private terminal: DBTTerminal,
-    private telemetry: TelemetryService,
+    protected commandProcessExecutionFactory: CommandProcessExecutionFactory,
+    protected pythonEnvironment: PythonEnvironment,
+    protected terminal: DBTTerminal,
+    protected telemetry: TelemetryService,
+    protected cwd: Uri,
   ) {}
 
   execute(command: DBTCommand, token?: CancellationToken): Promise<string> {
-    return this.executeCommand(command, token).completeWithTerminalOutput(
-      this.terminal,
-    );
+    const commandExecution = this.executeCommand(command, token);
+    if (command.logToTerminal) {
+      return commandExecution.completeWithTerminalOutput(this.terminal);
+    }
+    return commandExecution.complete();
   }
 
-  private executeCommand(
+  protected executeCommand(
     command: DBTCommand,
     token?: CancellationToken,
   ): CommandProcessExecution {
-    this.terminal.log(`> Executing task: ${command.getCommandAsString()}\n\r`);
+    if (command.logToTerminal) {
+      this.terminal.log(
+        `> Executing task: ${command.getCommandAsString()}\n\r`,
+      );
+    }
     this.telemetry.sendTelemetryEvent("dbtCommand", {
       command: command.getCommandAsString(),
     });
-    if (command.focus) {
+    if (command.logToTerminal && command.focus) {
```

Comment:
shouldn't be this as first line of this block? so the `executing task` log above will show show?